vllm/v1/worker/gpu_model_runner.py

```diff
@@ -952,6 +969,24 @@ def execute_model(
         sample_hidden_states = hidden_states[logits_indices]
         logits = self.model.compute_logits(sample_hidden_states, None)
 
+        # NOTE: We are currently broadcasting the bitmask
+        # to each worker
+        grammar_bitmask = scheduler_output.grammar_bitmask
+
+        # Apply guided decoding bitmasks if present
+        if grammar_bitmask is not None:
+            if len(self.input_batch.req_ids) < self.input_batch.max_num_reqs:
+                # The bitmask is pre-allocated for the maximum batch size.
+                # When the batch size is smaller, we need to resize the bitmask
+                # to match the batch size.
+                grammar_bitmask = grammar_bitmask[:len(self.input_batch.req_ids
+                                                       )]
```

Comment:
We should do this trimming on the scheduler side. Otherwise we're broadcasting extra data for no reason.